lib/vue/src/components/webcam/CWebcamModal.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <b-modal
+    ref="modal"
+    size="lg"
+    centered
+    :title="modalTitle"
+    :body-class="processingWebcam ? 'd-flex flex-column align-items-center' : 'p-0'"
+    @show="initializeWebcam"
+  >
+    <b-spinner
+      v-show="processingWebcam"
+      variant="primary"
+    />
+
+    <div
+      v-if="showErrorMessage"
+      class="bg-danger rounded p-2 m-2 text-white"
+    >
+      {{ cameraErrorMessage }}
+    </div>
+
+    <div
+      v-show="!processingWebcam"
+      class="embed-responsive embed-responsive-4by3"
+    >
+      <video
+        ref="video"
+        autoplay
+      />
+    </div>
+    <template #modal-footer>
+      <div class="d-flex align-items-center gap-2">
+        <b-button
+          ref="closeButton"
+          variant="light"
+          @click="closeCamera"
+        >
+          {{ cancelButtonLabel }}
+        </b-button>
+
+        <b-button
+          :disabled="processingWebcam"
+          variant="primary"
+          @click="() => (hasCapturedImage ? uploadCapturedImage() : capturePhoto())"
+        >
+          {{ hasCapturedImage
+            ? confirmButtonLabel
+            : captureButtonLabel }}
+        </b-button>
+      </div>
+    </template>
+  </b-modal>
+</template>
+
+<script>
+export default {
+  name: 'CWebcamModal',
+
+  props: {
+    modalTitle: {
+      type: String,
+      required: true,
+    },
+    cancelButtonLabel: {
+      type: String,
+      required: true,
+    },
+    confirmButtonLabel: {
+      type: String,
+      required: true,
+    },
+    captureButtonLabel: {
+      type: String,
+      required: true,
+    },
+    cameraErrorMessage: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      video: null,
+      stream: null,
+      capturedImage: null,
+      hasCapturedImage: false,
```

Comment:
Can't hasCapturedImage be computed that just checks if capturedImage exists? 
I don't see a point in duplicating that.